### PR TITLE
OLED support

### PR DESCRIPTION
**Pull request by noman1228**
Just env lib and main loop.

### CODE REVIEW

File: platformio.ini
Comment (MartinMueller2003) [inline platformio.ini:1]: Why did the header get removed?
Why are you adding an extra Platform type?

File: ESPixelStick/ESPixelStick.ino [hunk 1]
```diff
@@ -1,22 +1,3 @@
-/*
```
Comment (MartinMueller2003) [inline ESPixelStick/ESPixelStick.ino:1]: Please restore the headings 
Please restore the formatting. You removed much of the indenting. Please keep your changes to a minimum.

File: ESPixelStick/ESPixelStick.ino [hunk 2]
```diff
@@ -76,25 +59,37 @@ static void _u0_putc(char c){
 //  Globals
 //
 /////////////////////////////////////////////////////////
+// OLED
```
Comment (MartinMueller2003) [inline ESPixelStick/ESPixelStick.ino:65]: Please create an OLED file and move all of this to that file. Please make the entire file contents conditional (look at the temperature sensor setup). Then enable this in your platform.

File: ESPixelStick/ESPixelStick.ino [hunk 3]
```diff
@@ -76,25 +59,37 @@ static void _u0_putc(char c){
...
+#define SCREEN_HEIGHT 32    // OLED display height, in pixels
+#define OLED_RESET -1       // Reset pin # (or -1 if sharing Arduino reset pin)
+#define SCREEN_ADDRESS 0x3C ///< See datasheet for Address; 0x3D for 128x64, 0x3C for 128x32
+Adafruit_SSD1306 display(SCREEN_WIDTH, SCREEN_HEIGHT, &Wire, OLED_RESET);
 
 #ifdef ESPS_VERSION
 const String VERSION = STRING(ESPS_VERSION);
 #else
-const String VERSION = "4.x-dev";
+const String VERSION = "4.x-jmt";
```
Comment (MartinMueller2003) [inline ESPixelStick/ESPixelStick.ino:80]: Please restore this line. This is done using the platform name from platformio.ini

**Review by forkineye — CHANGES_REQUESTED** [review]
On top of what Martin has already noted, you deleted all the distribution and documentation files.

File: ESPixelStick/ESPixelStick.ino [hunk 4]
```diff
@@ -76,25 +59,37 @@ static void _u0_putc(char c){
...
 #endif
-
+int period = 5000;
```
Comment (MartinMueller2003) [inline ESPixelStick/ESPixelStick.ino:82]: These should not be here. Please move to the oled file.

File: ESPixelStick/ESPixelStick.ino [hunk 5]
```diff
@@ -133,11 +129,22 @@ void TestHeap(uint32_t Id)
 /** Arduino based setup code that is executed at startup. */
 void setup()
 {
+    if (!display.begin(SSD1306_SWITCHCAPVCC, SCREEN_ADDRESS))
+    {
+        for (;;)
+            ; // Don't proceed, loop forever
```
Comment (MartinMueller2003) [inline ESPixelStick/ESPixelStick.ino:138]: We never loop forever. If it does not work, then print out a message and keep going. The oled is not a critical resource for operation. If you want to for a reboot then set the reboot request flag and move on.

File: ESPixelStick/ESPixelStick.ino [hunk 6]
```diff
@@ -464,35 +471,34 @@ String serializeCore(bool pretty)
...
-    // DEBUG_START;
-/*
-    if(millis() > HeapTime)
-    {
-        DEBUG_V(String("Heap: ") + String(ESP.getFreeHeap()));
-        HeapTime += 5000;
-    }
-*/
-    FeedWDT ();
+
```
Comment (MartinMueller2003) [inline ESPixelStick/ESPixelStick.ino:494]: Please restore this code.

File: ESPixelStick/ESPixelStick.ino [hunk 7]
```diff
@@ -502,29 +508,41 @@ void loop()
...
     }
 
     if (ConfigSaveNeeded)
     {
-        FeedWDT ();
-        SaveConfig ();
+        FeedWDT();
+        SaveConfig();
+    }
+    if(millis() >= time_now + period){
```
Comment (MartinMueller2003) [inline ESPixelStick/ESPixelStick.ino:554]: Move this call to the oled files and make a call that is conditionally compiled.

File: ESPixelStick/src/ESPixelStick.h [hunk 8]
```diff
@@ -20,6 +20,7 @@
 
 #include <Arduino.h>
 
+
```
Comment (MartinMueller2003) [inline ESPixelStick/src/ESPixelStick.h:23]: Please dont change the formatting

File: ESPixelStick/src/GPIO_Defs.hpp [hunk 9]
```diff
@@ -128,10 +128,16 @@ typedef enum
 #   include "platformDefinitions/GPIO_Defs_ESP8266_ESPS_V3.hpp"
 #elif defined (BOARD_ESPS_ESP3DEUXQUATRO_DMX)
 #   include "platformDefinitions/GPIO_Defs_ESP32_ESP3DEUXQuattro_DMX.hpp"
+# BOARD_ESP32_DEVKITC_JMT
+#elif defined (BOARD_ESP32_DEVKITC_JMT)
+#   include "platformDefinitions/GPIO_Defs_ESP32_DevKitv1-JMT.hpp"
```
Comment (MartinMueller2003) [inline ESPixelStick/src/GPIO_Defs.hpp:133]: Is this really needed?

File: ESPixelStick/src/platformDefinitions/GPIO_Defs_ESP32_DevkitC.hpp [hunk 10]
```diff
@@ -32,17 +32,17 @@
 #define DEFAULT_SPI_DATA_GPIO   gpio_num_t::GPIO_NUM_16
 #define DEFAULT_SPI_CLOCK_GPIO  gpio_num_t::GPIO_NUM_17
 
-#define DEFAULT_I2C_SDA         gpio_num_t::GPIO_NUM_3
-#define DEFAULT_I2C_SCL         gpio_num_t::GPIO_NUM_5
+#define DEFAULT_I2C_SDA         gpio_num_t::GPIO_NUM_21
+#define DEFAULT_I2C_SCL         gpio_num_t::GPIO_NUM_22
 
-// File Manager
+// File Manager - FIXED JMT
```
Comment (MartinMueller2003) [inline ESPixelStick/src/platformDefinitions/GPIO_Defs_ESP32_DevkitC.hpp:38]: Please dont add such identifiers to the source code. Git blame provides this without cluttering up code.

File: SupportingDocs/FSEQ_Sequence_File_Format.txt [hunk 11]
```diff
@@ -1,64 +0,0 @@
-v1.0 spec
```
Comment (MartinMueller2003) [inline SupportingDocs/FSEQ_Sequence_File_Format.txt:1]: Why are these files being deleted?

**Comment by MartinMueller2003** [comment]
I have reached my single review comment limit. 
Please undo all of the format changes. They make doing a review difficult.
Please restore the missing files and restructure your oled support into an oled class. The class should be stand alone and technology independent. you can create a sub class for the specific oled you are implementing. Keen in mind that others may want to use other oled devices so your flavor should be a derived class from a pure virtual base class and the base class is used by the ino file.

**Comment by MartinMueller2003** [comment]
And please use useful commit comments.

**Comment by forkineye** [comment]
Closing this one due to the numerous issues.  Feel free to re-submit once they're corrected.

**Comment by noman1228** [comment]
Re-Forking and instituting minimal changes with conditional includes. Told you it was hacky.... Thanks for your patience and hard work!

**Pull request closed without merging**